kitsune/products/views.py

```diff
@@ -43,16 +44,19 @@ def product_landing(request, template, slug):
         else:
             latest_version = 0
 
-    # Don't show firefox download button at header at firefox product page
-    hide_fx_download = product.slug == 'firefox'
+    user_agent = request.META.get('HTTP_USER_AGENT', '')
+    browser = get_browser(user_agent)
+
+    # Only show firefox download button when product is firefox and browser is not Firefox
+    show_fx_download = (product.slug == 'firefox' and browser != 'Firefox')
```

Comment:
nit (and maybe not important): `get_browser` does case insensitive browser matching but here we only check for `Firefox`. Maybe it's worth using `lower()`

Reply:
[The tests](https://github.com/mozilla/kitsune/blob/master/kitsune/sumo/tests/test_utils.py#L188) check for `Firefox`, so I think we're good. But you raise a valid point - we're relying on the UA string capitalizing browser names, which is true now and probably will be in the future, but is still a potentially brittle check. I have to do some follow-up on this branch - I'll see about standardizing on lowercase for all instances.